### DISCUSSION TITLE
Summoning spells' names are too big for spell book

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#7623] [General]
We should tighten name to fit at least three lines.
![image](https://user-images.githubusercontent.com/55348946/149011991-7344e12b-9248-4552-a1ed-87ba51d57b18.png)

Two lines would be perfect......

**Comment by ihhub** [comment]
In the original game these spells are shown with an error as well:
![image](https://user-images.githubusercontent.com/19829520/82777714-822a4700-9e81-11ea-9e92-b3d4f795e153.png)

**Comment by ihhub** [comment]
Hi @Branikolog , actually we have 4 lines of text based on your given screenshot. I am not sure how we are going to handle that. Could you please provide a little input for this?

**Comment by Branikolog** [comment]
> Hi @Branikolog , actually we have 4 lines of text based on your given screenshot. I am not sure how we are going to handle that. Could you please provide a little input for this?

I offer you to take away word "Summon" and leave only ex: "Earth Elemental [15]".
Also note, that some spaces between rows are not the same. (Look Armageddon spell) So we can do some vertical squeezing. Also we can lessen space between icon and words under.

**Comment by ihhub** [comment]
The problem with names occurs in hero's spell book as well
![image](https://user-images.githubusercontent.com/19829520/85391022-3a7c0580-b57c-11ea-9301-8fba9e12af34.png)

**Comment by ihhub** [comment]
Moving this issue to 1.0 as the same happens in the original game.

**Comment by ihhub** [comment]
@Branikolog , please update this issue (the first post) as we don't display spell cost in mage guilds anymore.

**Comment by OogaBoogaCode** [comment]
I looked into the issue today, Going off of @Branikolog 's suggestion, I removed the 'Summon' word from the spell name. Few things to note:
I built the game on MacOS Monterey 12, M2 chip, And run the game at 960x600 fullscreen.
The Issue of text displaying behind the text seems to be tied to the game resolution you play at. I added two images with/without 'summon'
<img width="1470" alt="Screen Shot 2023-03-12 at 5 06 08 PM" src="https://user-images.githubusercontent.com/84861171/224573924-464973cf-172f-4e16-881d-5ea7ab073372.png">
<img width="1470" alt="Screen Shot 2023-03-12 at 5 04 44 PM" src="https://user-images.githubusercontent.com/84861171/224573933-89df429d-5332-44e9-9e3b-81b99a3e4f02.png">

**Comment by Kiiro-Yakumo** [comment]
I think at least "S." should be kept to keep with the proper spell name. Leaving it as "Air Elemental" (and same for the rest) makes it a bit odd, especially there are field versions of these spells to set as Guardian for the mines. Sure they are separated by the category, but still I think that clear "S." should be kept there.

**Comment by Branikolog** [comment]
Hello, @Kiiro-Yakumo 

> I think at least "S." should be kept to keep with the proper spell name. Leaving it as "Air Elemental" (and same for the rest) makes it a bit odd, especially there are field versions of these spells to set as Guardian for the mines. Sure they are separated by the category, but still I think that clear "S." should be kept there.

Actually field spells are called "Set xxx guardian". so spell names still would be different enough not to confuse players.
I still believe, that cutting off "summon" is the most safe variant to fix the issue.

**Comment by LeHerosInconnu** [comment]
Hello everyone,

I still think the word "Summon" should be part of the spell name.
The sprite of the spell represents an elemental, that's clear.
Now giving the spell name what is already represented by the spell sprite doesn't add anything.

An Air Elemental sprite with the spell name "Air Elemental" is a kind of redundancy.

But if the word "Summon" is used, it gives additional information.
I am in favor of using the spell names with the word "Summon":
"Summon Air", "Summon Earth", "Summon Fire" and "Summon Water".

An Air Elemental sprite with the spell name "Summon Air" seems more appropriate.

**Comment by zenseii** [comment]
> An Air Elemental sprite with the spell name "Summon Air" seems more appropriate.

I agree with this. We already have "Set Fire Guardian" and not "Set Fire Elemental Guardian".

**Comment by Kiiro-Yakumo** [comment]
In that case maybe "Summon Air E." ? Summon Air sounds a bit funny in itself. I have air around me, I will summon more!
That "E." would clearly add the fact that it's about Elemental.

@Branikolog - Hello, I admit I forgot for a moment that on map Elemental spells are separate entries.

**Comment by Branikolog** [comment]
Hi, @zenseii and @LeHerosInconnu !

>I agree with this. We already have "Set Fire Guardian" and not "Set Fire Elemental Guardian".

At least it's referring to a noun here: " Set Fire (who?) Guardian". In a case of battle we see: "Summon Fire" which is confusing for me, personally.
Having an icon with elemental and the name "Fire Elemental" is quite a representative combination, that evidently makes Fire Elemental appear or being placed on a battlefield. 

I would even accept a solution to show just "Summon Elemental" without an elemental type (since it refers to an icon with a picture of certain elemental type), rather than "Summon Fire" name for this spell. :)

That's my personal opinion, though.

Hi, @Kiiro-Yakumo !

>In that case maybe "Summon Air E." ? Summon Air sounds a bit funny in itself. I have air around me, I will summon more!
That "E." would clearly add the fact that it's about Elemental.

I'm not really sure... I can't remember if I saw abbreviations somewhere else in the game UI.

**Comment by LeHerosInconnu** [comment]
Hello everyone,

> Hi, @zenseii and @LeHerosInconnu !
> 
> > I agree with this. We already have "Set Fire Guardian" and not "Set Fire Elemental Guardian".
> 
> At least it's referring to a noun here: " Set Fire (who?) Guardian". In a case of battle we see: "Summon Fire" which is confusing for me, personally. Having an icon with elemental and the name "Fire Elemental" is quite a representative combination, that evidently makes Fire Elemental appear or being placed on a battlefield.
> 
> I would even accept a solution to show just "Summon Elemental" without an elemental type (since it refers to an icon with a picture of certain elemental type), rather than "Summon Fire" name for this spell. :)
> 
> That's my personal opinion, though.

We also need to think about where the spell name will be used without the spell graphic.

In fheroes2 (one example):

![Summon Elemental alone text 01](https://user-images.githubusercontent.com/43583994/234612091-0531436f-792d-4596-bd81-1b7eabb726d4.png)

My grading by level of understanding:
So "Air Elemental"... ...yes, what's up with "Air Elemental"?...
Let's try with "Milk Bottle"... ...yes, what's up with "Milk Bottle"?...
1) "Milk Bottle" ...so?...
2) "Summon Bottle" ...a bottle of what?...
3) "Summon Milk" ...like that?...
4) "Summon Milk Bottle" ...I get it!

And also, what will it do for comprehension in other languages?
The more I think about it, the more the name of the spell should remain in full.
I don't have any other ideas at the moment.
Maybe the title of the issue should be changed to:
"Spell book is too small for summoning spells' names" :D 👓 

> Hi, @Kiiro-Yakumo !
> 
> > In that case maybe "Summon Air E." ? Summon Air sounds a bit funny in itself. I have air around me, I will summon more!
> > That "E." would clearly add the fact that it's about Elemental.
> 
> I'm not really sure... I can't remember if I saw abbreviations somewhere else in the game UI.

There are some abbreviations in the game ("Upg.", "Damg"), but I think it is not appropriate in this case.

**Comment by ihhub** [comment]
I am converting this into a discussion as no solution was proposed.

**Comment by Branikolog** [comment]
So, The only solution I see for this issue now is to extend the spellbook:
![Extended spellbook](https://github.com/ihhub/fheroes2/assets/55348946/e93b2583-1072-4843-89ab-315ab9e568e0)
